homeassistant/components/luxtronik/__init__.py

```diff
@@ -0,0 +1,115 @@
+"""Support for Luxtronik heatpump controllers."""
+from datetime import timedelta
+import logging
+
+from luxtronik import LOGGER as LuxLogger, Luxtronik as Lux
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_PORT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+
+from .const import (
+    ATTR_PARAMETER,
+    ATTR_VALUE,
+    CONF_CALCULATIONS,
+    CONF_PARAMETERS,
+    CONF_SAFE,
+    CONF_VISIBILITIES,
+)
+
+LuxLogger.setLevel(level="WARNING")
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+DATA_LUXTRONIK = "DATA_LT"
+
+LUXTRONIK_PLATFORMS = ["binary_sensor", "sensor"]
+DOMAIN = "luxtronik"
+
+ENTITY_ID_FORMAT = DOMAIN + ".{}"
+
+SERVICE_WRITE = "write"
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_HOST): cv.string,
+                vol.Required(CONF_PORT, default=8889): cv.port,
+                vol.Optional(CONF_SAFE, default=True): cv.boolean,
+            }
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+SERVICE_WRITE_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_PARAMETER): cv.string,
+        vol.Required(ATTR_VALUE): vol.Any(cv.Number, cv.string),
+    }
+)
+
+
+def setup(hass, config):
+    """Set up the Luxtronik component."""
+    conf = config[DOMAIN]
+
+    host = conf.get(CONF_HOST)
+    port = conf.get(CONF_PORT)
+    safe = conf.get(CONF_SAFE)
+
+    luxtronik = Luxtronik(host, port, safe)
+    luxtronik.update()
+
+    hass.data[DATA_LUXTRONIK] = luxtronik
+
+    def write_parameter(service):
+        """Write a parameter to the Luxtronik heatpump."""
+        parameter = service.data.get(ATTR_PARAMETER)
+        value = service.data.get(ATTR_VALUE)
+        luxtronik.write(parameter, value)
+
+    hass.services.register(
+        DOMAIN, SERVICE_WRITE, write_parameter, schema=SERVICE_WRITE_SCHEMA
+    )
+
+    return True
+
+
+class Luxtronik:
+    """Handle all communication with Luxtronik."""
+
+    def __init__(self, host, port, safe=True):
```

Comment:
Remove `safe=True` since you already specify a default value in the respective schema.
```suggestion
    def __init__(self, host, port, safe):
```